docs/book/src/quinn/data-transfer.md

```diff
@@ -0,0 +1,165 @@
+# Data Transfer
+
+In the [previous chapter](set-up-connection.md) we characterized how to set up an [Endpoint][Endpoint]
+and then get access to a [NewConnection][NewConnection] instance.
+Now we will continue with the subject of sending data over this connection.
+
+## Multiplexing
+
+Multiplexing is the act of combining data from multiple streams into a single stream. 
+This can have a significant positive effect on the performance of the application. 
+QUIC it itself does not perform multiplexing instead it gives the programmer full control over this process. 
+  
+## Stream Types
+
+QUIC provides support for both stream and message based communication.
+Streams and messages can be initiated both on the client and server.
+
+| Type | Description | Reference |
+| :----- | :----- | :----- |
+| **Bidirectional Stream** | two way stream communication. | see [open_bi][open_bi] |
+| **Unidirectional Stream** | one way stream communication. | see [open_uni][open_uni] |
+| **Unreliable Messaging (extension)** | message based unreliable communication. | see [send_datagram][send_datagram] |
+
+## How to Use
+
+New streams can be created with the methods [open_bi][open_bi], [open_uni][open_uni] of type [Connection][Connection].
+An instance of this type, together with existing streams, can be found in the [connection][connection] field of [NewConnection].
+
+*Iterate over various streams*
```

Comment:
Here and below, I'm not sure `iterate` makes sense to me. What we're really doing here is waiting for incoming streams, which is more like "receiving" than "iterating", perhaps? Also the example code can be misleading. Also maybe the example should maybe make clear that you probably shouldn't use code like this because it would never get to receive bidirectional streams or datagrams, since it'd be waiting for unidirectional streams forever (until the peer closes its connection).

Reply:
Good one. I mainly added this example to show a very minimal usage of receiving from different streams. I will add a note for that. Also receiving would make more sense yea.